pkg/trace/agent/agent.go

```diff
@@ -236,9 +242,12 @@ func (a *Agent) Process(t pb.Trace) {
 		pt.Env = tenv
 	}
 
+	now = time.Now()
+
 	go func() {
 		defer watchdog.LogOnPanic()
 		a.ServiceExtractor.Process(pt.WeightedTrace)
+		metrics.Timing("datadog.trace_agent.receiver.process.service_extractor.time", time.Since(now), nil, 1)
```

Comment:
The `Process` function now has 4 new metrics. This means that if we were to process 2000 TPS it would result in 8000 UDP calls per second, not to speak of other internals that run when timing. I don't think we want this, unless planning to test in a non-production environment.